ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/tls/TlsConfiguration.java

```diff
@@ -0,0 +1,96 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.ethereum.api.tls;
+
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.Optional;
+
+import io.vertx.core.net.PfxOptions;
+import io.vertx.core.net.TrustOptions;
+import org.apache.tuweni.net.tls.VertxTrustOptions;
+
+public class TlsConfiguration {
+  private final Path keyStorePath;
+  private final String keyStorePassword;
+  private final Path knownClientsFile;
+  private TrustOptions trustOptions;
+  private PfxOptions pfxKeyCertOptions;
+
+  private TlsConfiguration(
+      final Path keyStorePath, final String keyStorePassword, final Path knownClientsFile) {
+    Objects.requireNonNull(keyStorePath, "Key Store Path must not be null");
+    Objects.requireNonNull(keyStorePassword, "Key Store password must not be null");
+    this.keyStorePath = keyStorePath;
+    this.keyStorePassword = keyStorePassword;
+    this.knownClientsFile = knownClientsFile;
+  }
+
+  private void init() {
+    this.pfxKeyCertOptions =
+        new PfxOptions().setPath(keyStorePath.toString()).setPassword(keyStorePassword);
+    try {
+      this.trustOptions =
+          Optional.ofNullable(knownClientsFile)
+              .map(VertxTrustOptions::whitelistClients)
+              .orElse(null);
+    } catch (RuntimeException exception) {
+      throw new TlsConfigurationException(exception.getMessage());
+    }
+  }
+
+  public PfxOptions getPfxKeyCertOptions() {
+    return pfxKeyCertOptions;
+  }
+
+  public Optional<TrustOptions> getTrustOptions() {
+    return Optional.ofNullable(trustOptions);
+  }
+
+  public static final class TlsConfigurationBuilder {
+    private Path keyStorePath;
+    private String keyStorePassword;
+    private Path knownClientsFile;
+
+    private TlsConfigurationBuilder() {}
+
+    public static TlsConfigurationBuilder aTlsConfiguration() {
+      return new TlsConfigurationBuilder();
+    }
+
+    public TlsConfigurationBuilder withKeyStorePath(final Path keyStorePath) {
+      this.keyStorePath = keyStorePath;
+      return this;
+    }
+
+    public TlsConfigurationBuilder withKeyStorePassword(final String keyStorePassword) {
+      this.keyStorePassword = keyStorePassword;
+      return this;
+    }
+
+    public TlsConfigurationBuilder withKnownClientsFile(final Path knownClientsFile) {
+      this.knownClientsFile = knownClientsFile;
+      return this;
+    }
+
+    public TlsConfiguration build() {
```

Comment:
probalby worth doing all the work in this build() function rather than the init - thus the config object doens't need to effectively duplicate the data.

Reply:
👍